### PR TITLE
feat: Add app-registry initial

### PR DESCRIPTION
**Pull request by toschdev**
Adds the App Registry within `ignite/apps/_ignite-app-registry` directory. With an example for the `explorer` Ignite App.

It's expected that in the `_ignite-app-registry`, each app has it's own directory with an including `app.json` in the format: `appname/app.json`.

Here's the `schema.app.json`:

```json
{
    "appName": "Your App Name",
    "appDescription": "A brief description of what your app does.",
    "latestRelease": {
      "version": "x.y.z",
      "date": "YYYY-MM-DD",
      "releaseNotesUrl": "URL to the release notes or changelog"
    },
    "igniteCompatibility": {
      "lastTestedVersion": "ignite vX.Y.Z"
    },
    "cosmosSdkCompatibility": {
      "lastWorkedVersion": "vX.Y.Z"
    },
    "features": [
      "Feature 1",
      "Feature 2",
      "Additional features..."
    ],
    "wasm": "Indicate if your app uses wasm (yes/no)",
    "authors": [
      {
        "name": "Author Name",
        "email": "email@example.com",
        "website": "Optional author or company website"
      }
    ],
    "repository": {
      "url": "URL to the app's repository",
      "type": "e.g., GitHub, GitLab"
    },
    "documentationUrl": "URL to the app's documentation",
    "license": "License type, e.g., MIT, Apache 2.0",
    "keywords": ["keyword1", "keyword2", "Useful for search and categorization"],
    "supportedPlatforms": ["platform1", "platform2", "e.g., osmosis, cosmoshub"],
    "socialMedia": {
      "twitter": "Optional Twitter handle",
      "telegram": "Optional Telegram group",
      "discord": "Optional Discord server",
      "reddit": "Optional Reddit page"
    },
    "donations": {
      "cryptoAddresses": {
        "cosmos": "cosmos1...",
        "otherSupportedCryptos": "address"
      },
      "fiatDonationLinks": "URL to Patreon, Ko-fi, etc."
    }
}
```

### CODE REVIEW

File: _ignite-app-registry/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+## **How to Add Your Ignite App**
+
+To add your app to the Ignite App Registry, follow these steps:
+
+1. **Fork the Repository**: Start by forking the **`ignite/apps`** repository to your own GitHub account.
+2. **Create a New Directory within `*_ignite-app-registy`**:* Inside the repository and inside the directory `_ignite_app_registry`, create a new directory named after your app. Ensure the name is unique and descriptive.
+3. **Add Your `app.json` File**: In your app's directory, add an **`app.json`** file. This file should contain all the relevant details about your app according to the template provided below.
+4. **Submit a Pull Request**: Once you've added your app's directory and **`app.json`** file, submit a pull request to the original **`ignite/apps`** repository. Your pull request will be reviewed by the  maintainers, and once approved, your app will be listed in the registry.
+
+## **`app.json` File Structure**
```
Comment (julienrbrt) [inline _ignite-app-registry/README.md:16]: nit, shall we use snake_case instead of camelCase?

File: _ignite-app-registry/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    "authors": [
+      {
+        "name": "Author Name",
+        "email": "email@example.com",
+        "website": "Optional author or company website"
+      }
+    ],
+    "repository": {
+      "url": "URL to the app's repository",
+      "type": "e.g., GitHub, GitLab"
```
Comment (julienrbrt) [inline _ignite-app-registry/README.md:50]: Why is that needed? We can infer that from the url.

File: _ignite-app-registry/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    },
+    "documentationUrl": "URL to the app's documentation",
+    "license": "License type, e.g., MIT, Apache 2.0",
+    "keywords": ["keyword1", "keyword2", "Useful for search and categorization"],
+    "supportedPlatforms": ["platform1", "platform2", "e.g., osmosis, cosmoshub"],
```
Comment (julienrbrt) [inline _ignite-app-registry/README.md:55]: I'm not sure if this makes sense

File: _ignite-app-registry/explorer/app.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{
+    "appName": "explorer",
+    "appDescription": "Shows real time feed of your blockchain.",
+    "latestRelease": {
+      "version": "0.2",
+      "date": "2024-02-27",
+      "releaseNotesUrl": "https://github.com/ignite/apps/releases/tag/explorer%2Fv0.2.0"
+    },
+    "igniteCompatibility": {
+      "lastTestedVersion": "ignite v28.2.0"
```
Comment (julienrbrt) [inline _ignite-app-registry/explorer/app.json:10]: I think it should just be the version

**Review by julienrbrt — DISMISSED** [review]
Lgtm, just some suggestions but feel free to dismiss

File: _ignite-app-registry/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+## **Overview**
+
+The Ignite App Registry is a directory designed to catalog applications built within the Ignite Apps ecosystem. It aims to facilitate discovery, collaboration, and information sharing among developers and users. By submitting your app to the registry, you make it easier for the community to learn about and engage with your project.
+
+## **How to Add Your Ignite App**
+
+To add your app to the Ignite App Registry, follow these steps:
+
+1. **Fork the Repository**: Start by forking the **`ignite/apps`** repository to your own GitHub account.
```
Comment (salmad3) [inline _ignite-app-registry/README.md:11]: The prompts for each point are a bit redundant with the points themselves.

I think we should keep either or.

Only prompts (as they are quite intuitive):

```
1. Fork the Repository
2. Create a New Directory within `_ignite-app-registy`
3. Add Your `app.json` file
4. Submit a Pull Request
```

Or 

```
1. Start by forking the **`ignite/apps`** repository to your own GitHub account.
2. Inside the repository and inside the directory `_ignite_app_registry`, create a new directory named after your app. Ensure the name is unique and descriptive.
3. In your app's directory, add an **`app.json`** file. This file should contain all the relevant details about your app according to the template provided below.
4. Once you've added your app's directory and **`app.json`** file, submit a pull request to the original **`ignite/apps`** repository. Your pull request will be reviewed by the maintainers, and once approved, your app will be listed in the registry.
```

File: _ignite-app-registry/README.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      {
+        "name": "Author Name",
+        "email": "email@example.com",
+        "website": "Optional author or company website"
+      }
+    ],
+    "repository": {
+      "url": "URL to the app's repository",
+      "type": "e.g., GitHub, GitLab"
+    },
```
Comment (salmad3) [inline _ignite-app-registry/README.md:51]: What about quick install instructions? Something like (or a simpler version of):

```json
"installation": {
  "instructions": "Quick install instructions or a brief overview.",
  "command": "ignite app install <your-app-name>",
  "url": "URL to detailed installation guide"
},
```

File: _ignite-app-registry/README.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    "cosmosSdkCompatibility": {
+      "lastWorkedVersion": "vX.Y.Z"
+    },
+    "features": [
+      "Feature 1",
+      "Feature 2",
+      "Additional features..."
+    ],
+    "wasm": "Indicate if your app uses wasm (yes/no)",
```
Comment (salmad3) [inline _ignite-app-registry/README.md:40]: An extension of this could be if the app has dependancies on other apps: Something like:

```json
"dependencies": {
  "ignite_apps": ["ignite app name 1", "ignite app name 2"],
  "module": ["module name 1", "module name 2"]
},
```

File: _ignite-app-registry/README.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
+# **Ignite App Registry**
+
+## **Overview**
+
+The Ignite App Registry is a directory designed to catalog applications built within the Ignite Apps ecosystem. It aims to facilitate discovery, collaboration, and information sharing among developers and users. By submitting your app to the registry, you make it easier for the community to learn about and engage with your project.
+
```
Comment (salmad3) [inline _ignite-app-registry/README.md:6]: I recommend adding a disclaimer to clarify that inclusion in the registry is not an official endorsement or recognition. The disclaimer should emphasize that users should perform their own due diligence when using apps from the registry.

File: _ignite-app-registry/README.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+To add your app to the Ignite App Registry, follow these steps:
+
+1. **Fork the Repository**: Start by forking the **`ignite/apps`** repository to your own GitHub account.
+2. **Create a New Directory within `*_ignite-app-registy`**:* Inside the repository and inside the directory `_ignite_app_registry`, create a new directory named after your app. Ensure the name is unique and descriptive.
+3. **Add Your `app.json` File**: In your app's directory, add an **`app.json`** file. This file should contain all the relevant details about your app according to the template provided below.
+4. **Submit a Pull Request**: Once you've added your app's directory and **`app.json`** file, submit a pull request to the original **`ignite/apps`** repository. Your pull request will be reviewed by the  maintainers, and once approved, your app will be listed in the registry.
+
+## **`app.json` File Structure**
+
+Below is the template for the **`app.json`** file. Replace each placeholder with the specific details of your app.
```
Comment (salmad3) [inline _ignite-app-registry/README.md:18]: It would be helpful here to have complimentary instructions on which entries are required and which are optional (maybe in a dropdown), and for the ones that aren't specified, which keywords to use (e.g., "null")

Such as for the social media object:
```json
"socialMedia": {
  "x": "app_x_handle",
  "telegram": null,
  "discord": null,
  "reddit": null
}
```

Or simply omitting the object.

File: _ignite-app-registry/README.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    "repository": {
+      "url": "URL to the app's repository",
+      "type": "e.g., GitHub, GitLab"
+    },
+    "documentationUrl": "URL to the app's documentation",
+    "license": "License type, e.g., MIT, Apache 2.0",
+    "keywords": ["keyword1", "keyword2", "Useful for search and categorization"],
+    "supportedPlatforms": ["platform1", "platform2", "e.g., osmosis, cosmoshub"],
+    "socialMedia": {
+      "twitter": "Optional Twitter handle",
```
Comment (salmad3) [inline _ignite-app-registry/README.md:57]:
```suggestion
      "x": "Optional X handle",
```

**Review by salmad3 — COMMENTED** [review]
Great @toschdev, left some suggestions, primarily nits and enhancements.

File: _ignite-app-registry/README.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+        "email": "email@example.com",
+        "website": "Optional author or company website"
+      }
+    ],
+    "repository": {
+      "url": "URL to the app's repository",
+      "type": "e.g., GitHub, GitLab"
+    },
+    "documentationUrl": "URL to the app's documentation",
+    "license": "License type, e.g., MIT, Apache 2.0",
```
Comment (Pantani) [inline _ignite-app-registry/README.md:53]: maybe the license link?

File: _ignite-app-registry/explorer/app.json [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    ],
+    "repository": {
+      "url": "https://github.com/ignite/apps/tree/explorer/v0.2.0/explorer",
+      "type": "GitHub"
+    },
+    "documentationUrl": "https://github.com/cosmos/gex",
+    "license": "MIT",
+    "keywords": ["explorer", "cli", "cosmos-sdk", "ignite app"],
+    "supportedPlatforms": ["macos", "windows", "linux", "cosmos", "osomosis", "evmos", "celestia", "dymension"],
```
Comment (Pantani) [inline _ignite-app-registry/explorer/app.json:35]:
```suggestion
    "supportedPlatforms": ["macos", "windows", "linux"],
```

**Comment by toschdev** [comment]
Thanks for all the comments, will try to address them. I really prefer commit suggestions because these are a lot of ideas to unpack.

File: _ignite-app-registry/README.md (hunk 9 above)
Comment (jeronimoalbi) [reply _ignite-app-registry/README.md:18]: Maybe too much but we could also define a [JSON schema](https://json-schema.org/) which would, in a way, document the format and rules of the app schema. This adds the plus of being able to validate app schemas using a CI workflow I think.

@toschdev if we want to implement this idea I could write the JSON schema or we could do it as pair programming if you want.

**Comment by jeronimoalbi** [comment]
> Thanks for all the comments, will try to address them. I really prefer commit suggestions because these are a lot of ideas to unpack.

Adding a couple more, just for the record :)

I would not use folders for the different app schemas unless we would have extra files per app, like pictures or documents, which might be better to avoid. That means all app schemas would be available at the same level under `./_ignite-app-registry`. Each app schema could follow the naming `foo_nam.app.json` (or YAML?).

An alternative name for `schema.app.json` could be `registry.json`.

If we decide on the [JSON schema idea](https://github.com/ignite/apps/pull/65#discussion_r1531792856) we could defined the JSON schema as `schema.json`, so we would have `registry.json` and `schema.json`

I would keep the schema as simple as possible with the minimum set of features we know are useful right now and maybe I would introduce more features "on demand" or as they are required.

Inspiration could be found in other well stablished schemas:
- [package.json](https://docs.npmjs.com/cli/v10/configuring-npm/package-json)
- [pyproject.toml](https://python-poetry.org/docs/pyproject/)

Nit: Maybe we could just use `registry` for the folder name instead of `_ignite-app-registry` so the URL is more semantic `https://github.com/ignite/apps/registry`

File: _ignite-app-registry/README.md (hunk 1 above)
Comment (toschdev) [reply _ignite-app-registry/README.md:16]: JSON schema suggests camelCase for our usecase: https://json-schema.org/learn/getting-started-step-by-step

File: _ignite-app-registry/explorer/app.json (hunk 4 above)
Comment (toschdev) [reply _ignite-app-registry/explorer/app.json:10]: We'll use version like defined in: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#dependencies

File: _ignite-app-registry/README.md (hunk 6 above)
Comment (toschdev) [reply _ignite-app-registry/README.md:51]: I'd consider this premature optimisation, thanks for the suggestion

File: _ignite-app-registry/README.md (hunk 7 above)
Comment (toschdev) [reply _ignite-app-registry/README.md:40]: We have dependencies now. Thanks

File: _ignite-app-registry/README.md (hunk 8 above)
Comment (toschdev) [reply _ignite-app-registry/README.md:6]: Done

File: _ignite-app-registry/README.md (hunk 9 above)
Comment (toschdev) [reply _ignite-app-registry/README.md:18]: Yes, using JSON schema is the right approach IMO

File: _ignite-app-registry/README.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+3. **Add Your `app.json` File**: In your app's directory, add an **`app.json`** file. This file should contain all the relevant details about your app according to the template provided below.
+4. **Submit a Pull Request**: Once you've added your app's directory and **`app.json`** file, submit a pull request to the original **`ignite/apps`** repository. Your pull request will be reviewed by the  maintainers, and once approved, your app will be listed in the registry.
+
+## **`app.json` File Structure**
+
+Below is the template for the **`app.json`** file. Replace each placeholder with the specific details of your app.
+
+```json
+{
+    "appName": "Your App Name",
```
[the remark this answers is not in the document]
Comment (toschdev) [reply _ignite-app-registry/README.md:22]: We need to have a check in CI for this

File: _ignite-app-registry/README.md [hunk 14]
```diff
@@ -8,14 +8,14 @@ The Ignite App Registry is a directory designed to catalog applications built wi
 
 To add your app to the Ignite App Registry, follow these steps:
```
Comment (salmad3) [inline _ignite-app-registry/README.md:9]: [Still redundant](https://github.com/ignite/apps/pull/65#discussion_r1509617234)

**Review by salmad3 — DISMISSED** [review]

File: _ignite-app-registry/README.md (hunk 7 above)
Comment (salmad3) [reply _ignite-app-registry/README.md:40]: The suggestion was on app dependencies.

File: _ignite-app-registry/README.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+        "website": "Optional author or company website"
+      }
+    ],
+    "repository": {
+      "url": "URL to the app's repository",
+      "type": "e.g., GitHub, GitLab"
+    },
+    "documentationUrl": "URL to the app's documentation",
+    "license": {
+      "name": "MIT",
```
[the remark this answers is not in the document]
Comment (toschdev) [reply _ignite-app-registry/README.md:54]: No, it's an example

File: _ignite-app-registry/README.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+
+## **Overview**
+
+The Ignite App Registry is a directory designed to catalog applications built within the Ignite Apps ecosystem. It aims to facilitate discovery, collaboration, and information sharing among developers and users. By submitting your app to the registry, you make it easier for the community to learn about and engage with your project.
+
+## **How to Add Your Ignite App**
+
+To add your app to the Ignite App Registry, follow these steps:
+
+1. **Fork the Repository**: Start by forking the `ignite/apps` repository to your own GitHub account.
```
[the remark this answers is not in the document]
Comment (toschdev) [reply _ignite-app-registry/README.md:11]: No, you need to fork, you cannot work with a "clone" or copy

File: _ignite-app-registry/README.md [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+2. **Create a New Directory within `_ignite-app-registy`: Inside the repository and inside the directory `_ignite_app_registry`, create a new directory named after your app. Ensure the name is unique and descriptive.
+3. **Add Your `app.json` File**: In your app's directory, add an `app.json` file. This file should contain all the relevant details about your app according to the template provided below.
+4. **Submit a Pull Request**: Once you've added your app's directory and `app.json` file, submit a pull request to the original `ignite/apps` repository. Your pull request will be reviewed by the  maintainers, and once approved, your app will be listed in the registry.
```
[the remark this answers is not in the document]
Comment (toschdev) [reply _ignite-app-registry/README.md:14]: I think being direct "Submit a Pull Request" describes this better.

File: _ignite-app-registry/README.md (hunk 7 above)
Comment (toschdev) [reply _ignite-app-registry/README.md:40]: We've decided against it.

**Comment by salmad3** [comment]
@jeronimoalbi @Pantani @julienrbrt bumping again, please re-review so that we can add the initial version.

File: _ignite-app-registry/README.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+
+## **`app.json` File Structure**
+
+Below is the template for the **`app.json`** file. Replace each placeholder with the specific details of your app.
+
+```json
+{
+    "appName": "Your App Name",
+    "appDescription": "A brief description of what your app does.",
+    "latestRelease": {
```
[the remark this answers is not in the document]
Comment (Pantani) [reply _ignite-app-registry/README.md:24]: Are there any updates on it? will we still keep this info?

**Review by julienrbrt — APPROVED** [review]
ACK. we probably can always increment on this later.

**Review by salmad3 — APPROVED** [review]

File: _ignite-app-registry/README.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (toschdev) [reply _ignite-app-registry/README.md:24]: It's already gone
